File: Arrays-And-Hashing/Contains-Duplicate.py
# 🏏 Top 150 NotOut - Contains Duplicate
# Author: Neeraj Singh

def hasDuplicate(nums):
    seen = set()
    for num in nums:
        if num in seen:
            return True
        seen.add(num)
    return False

# Taking user input
n = int(input("Enter the value : "))
nums = list(map(int, input("Enter numbers separated by space: ").split()))

# Displaying result
if hasDuplicate(nums):
    print("✅ Yes, the array contains duplicates.")
else:
    print("🎯 No, all elements are unique.")

File: Arrays-And-Hashing/contains-Duplicate.java
// 🏏 Top 150 NotOut - Contains Duplicate
// Author: Neeraj Singh

import java.util.*;

public class ContainsDuplicate {
    public static boolean hasDuplicate(int[] nums) {
        Set<Integer> seen = new HashSet<>();
        for (int num : nums) {
            if (seen.contains(num)) {
                return true;
            }
            seen.add(num);
        }
        return false;
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter numbers separated by space: ");
        String[] input = sc.nextLine().split(" ");
        int[] nums = new int[input.length];

        for (int i = 0; i < input.length; i++) {
            nums[i] = Integer.parseInt(input[i]);
        }

        if (hasDuplicate(nums)) {
            System.out.println("✅ Yes, the array contains duplicates.");
        } else {
            System.out.println("🎯 No, all elements are unique.");
        }

        sc.close();
    }
}
